packages/@aws-cdk/aws-lambda/test/test.log-retention.ts

```diff
@@ -93,5 +93,26 @@ export = {
 
     test.done();
 
+  },
+
+  'with RetentionPeriod set to Infinity'(test: Test) {
+    const stack = new cdk.Stack();
+
+    new LogRetention(stack, 'MyLambda', {
+      logGroupName: 'group',
+      retention: logs.RetentionDays.INFINITE,
+    });
+
+    expect(stack).to(haveResource('Custom::LogRetention',
+      (resourceProps: any, inspection: InspectionFailure) => {
+        if (resourceProps.RetentionInDays !== undefined) {
+          inspection.failureReason = 'RetentionInDays was present when it was not expected';
+          return false;
+        }
+        return true;
+      }
+    ));
```

Comment:
wasn't it possible to simply assert against the synthesized output? Why do we need this inspection callback?

Reply:
How do you test for the absence of a property from the synthesized output?

The template is huuge to use `matchTemplate()` while `haveResource()` can only test for the presence of AFAIK. Is there a assertion matcher or flag I'm missing?

Reply:
Oh... maybe you can use “not(haveResource”

Reply:
Still doesn't quite work. 

With `not(haveResource(`, I can check that it's not a specific value but the correct assertion is that it's not any value.

The bug that this is fixing set `RetentionInDays` to 9999.